webui/react/src/pages/F_ExpList/glide-table/ColumnPickerMenu.tsx

```diff
@@ -0,0 +1,234 @@
+  );
+  const metricsColumns: Record<string, boolean> = Form.useWatch(V1LocationType.VALIDATIONS, form);
+
+  const allFormColumns = useMemo(
+    () => ({ ...generalColumns, ...hyperparametersColumns, ...metricsColumns }),
+    [generalColumns, hyperparametersColumns, metricsColumns],
+  );
+
+  useEffect(() => {
+    if (Object.keys(allFormColumns).length === 0) return;
+    /* eslint-disable @typescript-eslint/no-unused-vars */
+    setVisibleColumns(
+      Object.entries(allFormColumns)
+        .filter(([_, checked]) => checked)
+        .map(([column, _]) => column),
+    );
+    /* eslint-enable @typescript-eslint/no-unused-vars */
+    // eslint-disable-next-line react-hooks/exhaustive-deps
+  }, [allFormColumns, setVisibleColumns]);
+
+  const handleShowSuggested = useCallback(() => {
+    const defaultCols: Set<string> = new Set(defaultExperimentColumns);
+
+    const newGeneral = { ...generalColumns };
+    for (const col of Object.keys(newGeneral)) {
+      newGeneral[col] = defaultCols.has(col);
+    }
+    form.setFieldValue(V1LocationType.EXPERIMENT, newGeneral);
+
+    const newHyperparameters = { ...hyperparametersColumns };
+    for (const col of Object.keys(newHyperparameters)) {
+      newHyperparameters[col] = defaultCols.has(col);
+    }
+    form.setFieldValue(V1LocationType.HYPERPARAMETERS, newHyperparameters);
+
+    const newMetrics = { ...metricsColumns };
+    for (const col of Object.keys(newMetrics)) {
+      newMetrics[col] = defaultCols.has(col);
+    }
+    form.setFieldValue(V1LocationType.VALIDATIONS, newMetrics);
+  }, [form, generalColumns, hyperparametersColumns, metricsColumns]);
+
+  const tabFilteredColumnsAllChecked = useMemo(() => {
+    return totalColumns
+      .filter((col) => filteredColumns.has(col.column) && col.location === activeColumnTab)
+      .map((col) => allFormColumns[col.column])
+      .every((col) => col === true);
+  }, [activeColumnTab, allFormColumns, filteredColumns, totalColumns]);
+
+  const handleShowHideAll = useCallback(() => {
+    const currentTabColumns = Object.fromEntries(
+      totalColumns
+        .filter((col) => isEqual(col.location, activeColumnTab) && col.column in allFormColumns)
+        .map((col) => [col.column, allFormColumns[col.column]]),
+    );
+    const filteredTabColumns: Record<string, boolean> = totalColumns
+      .filter((col) => filteredColumns.has(col.column) && col.location === activeColumnTab)
+      .reduce((acc, col) => ({ ...acc, [col.column]: !tabFilteredColumnsAllChecked }), {});
+
+    form.setFieldValue(activeColumnTab, Object.assign(currentTabColumns, filteredTabColumns));
+  }, [
+    activeColumnTab,
+    allFormColumns,
+    filteredColumns,
+    form,
+    tabFilteredColumnsAllChecked,
+    totalColumns,
+  ]);
+
+  const tabContent = useCallback(
+    (tab: V1LocationType) => {
+      return (
+        <div>
+          <Form.Item name="column-search">
+            <Input allowClear placeholder="Search" ref={searchRef} />
```

Comment:
I tried this and did run into losing the focus on tab switching. I actually have the tabs using `forceRender: true` so that inactive tab panes stay present, this is required for the form items on inactive tabs to continue existing.

Reply:
Actually, after further consideration I think it's fine if focus doesn't persist on tab switching, so I went ahead and used autoFocus.